packages/contracts-bedrock/scripts/Deploy.s.sol

```diff
@@ -554,7 +552,7 @@ contract Deploy is Deployer {
         // are always proxies.
         Types.ContractSet memory contracts = _proxiesUnstrict();
         contracts.OptimismMintableERC20Factory = address(factory);
-        ChainAssertions.checkOptimismMintableERC20Factory(contracts);
+        ChainAssertions.checkOptimismMintableERC20Factory({ _contracts: contracts, _isInitialized: false });
```

Comment:
Similar to my other comment, I think that `_isProxy: false` would be more accurate. 

I would say this arg key/value is wrong since the implementation is actually initialized by the constructor.

Reply:
agreed with the point about key/value being wrong since the implementation is actually initialized by the constructor. 

however, with respect to the first point, the call at deployment to `checkOptimismMintableERC20Factory` would fail because, in a way analogous to checkL1ERC721Bridge with L1ERC721Bridge (since `_isProxy` already exists for that case), the storage variables would be checked against dependent contracts at a time [they may point to null values](https://github.com/ethereum-optimism/optimism/blob/13c86c8c5d843265826b2a381409c28d2dff3b5a/packages/contracts-bedrock/src/L1/L1ERC721Bridge.sol#L32), even if `_isProxy: false`:

https://github.com/ethereum-optimism/optimism/blob/9b513d7c18048bceeb132c1796cf1975fd332e91/packages/contracts-bedrock/scripts/ChainAssertions.sol#L163-L175

Reply:
maybe a better call would be to rename that parameter to `isDeploying`? doesn't feel great for some reason though... do you have any better ideas @maurelian? 🤔

Reply:
I still don't understand my `_isProxy: false` isn't applicable when checking the implementation. The state of the implementation will never change whether initialized or not. 

Is your point that we might want to run these checks on the proxy in both its initialized and uninitialized states?

Reply:
exactly yes

Reply:
I see, though we currently are not using it in that way anywhere.

Regardless, since we agree on this point: 

> agreed with the second point about key/value being wrong since the implementation is actually initialized by the constructor.

IMO a reasonable path forward would be to address that, and see what that leads to.

Reply:
sounds good! I think thats a good idea. fwiw, this discussion could be useful for context: https://github.com/ethereum-optimism/optimism/pull/8631#discussion_r1428267234

Reply:
also thanks for all your feedback @maurelian! :) it really is greatly appreciated